mangaki/mangaki/utils/anidb.py

```diff
@@ -45,98 +40,43 @@ def search(self, q):
     results = []
     animetitles = BeautifulSoup(r.text, 'xml').animetitles
     for anime in animetitles.find_all('anime'):
-      results.append(Anime({
-        'id': int(anime['aid']),
-        'title': str(anime.find('title', attrs={'type': "official"}).string),
-        'picture': "http://img7.anidb.net/pics/anime/" + str(anime.find('picture').string),
-      }, partial=True, updater=lambda: self.get(anime.id)))
-
+      results.append(
```

Comment:
It appears to me this is a good candidate to transform this function into a generator, if `animetitles.find_all` is also a generator, of course.

Otherwise, I'm wondering if it'd be more optimal to use a `dict` to store animes by IDs, anyway, this is okay.